modules/novatiqIdSystem.js

```diff
@@ -47,18 +47,94 @@ export const novatiqIdSubmodule = {
     logInfo('NOVATIQ Sync request used sourceid param: ' + srcId);
+    logInfo('NOVATIQ Sync request Shared ID: ' + sharedId);
 
     let partnerhost;
     partnerhost = window.location.hostname;
     logInfo('NOVATIQ partner hostname: ' + partnerhost);
 
     const novatiqId = snowflakeId();
-    const url = 'https://spadsync.com/sync?sptoken=' + novatiqId + '&sspid=' + srcId + '&ssphost=' + partnerhost;
+
+    let url = 'https://spadsync.com/sync?sptoken=' + novatiqId + '&sspid=' + srcId + '&ssphost=' + partnerhost;
+
+    // for testing
+    let sharedStatus = 'Not Found';
+
+    // append on the shared ID if we have one
+    if (sharedId != null) {
+      url = url + '&sharedId=' + sharedId;
+      sharedStatus = 'Found';
+    }
+
     ajax(url, undefined, undefined, { method: 'GET', withCredentials: false });
 
     logInfo('NOVATIQ snowflake: ' + novatiqId);
-    return { 'id': novatiqId }
+    return { 'id': novatiqId,
+      'sharedStatus': sharedStatus }
+  },
+
+  useSharedId(configParams) {
+    return typeof configParams.useSharedId != 'undefined' && configParams.useSharedId === true;
+  },
+
+  getFromLocalStorage(cookieOrStorageID) {
+    return window.localStorage.getItem(cookieOrStorageID);
+  },
+
+  getFromSessionStorage(cookieOrStorageID) {
+    return window.sessionStorage.getItem(cookieOrStorageID);
+  },
+
+  // return null if we aren't supposed to use one or we are but there isn't one present
+  getSharedId(configParams) {
+    let sharedId = null;
+    if (this.useSharedId(configParams)) {
+      let cookieOrStorageID = '_pubcid';
+
+      // Has the cookieOrStorageID been redefined?
+      if (typeof configParams.sharedIdName != 'undefined' && configParams.sharedIdName != null && configParams.sharedIdName != '') {
+        cookieOrStorageID = configParams.sharedIdName;
+        logInfo('NOVATIQ sharedID name redefined: ' + cookieOrStorageID);
+      }
+
+      // Check local storage first
+      sharedId = this.getFromLocalStorage(cookieOrStorageID);
+      if (sharedId == null) {
+        // next check session storage
+        sharedId = this.getFromSessionStorage(cookieOrStorageID);
+
+        // OK check cookies
+        if (sharedId == null) {
+          sharedId = this.getCookieValue(cookieOrStorageID);
+          logInfo('NOVATIQ retrieved sharedID from cookies: ' + sharedId);
+        } else {
+          logInfo('NOVATIQ retrieved sharedID from sessionstorage: ' + sharedId);
+        }
+      } else {
+        logInfo('NOVATIQ retrieved sharedID from localstorage: ' + sharedId);
+      }
+
+      // sanity check
+      if (sharedId == undefined || sharedId == '') {
+        logInfo('NOVATIQ invalid sharedID, resetting to null');
+        sharedId = null;
+      }
+    }
+
+    return sharedId;
+  },
+
+  getCookieValue(name) {
+    let res = document.cookie.match('(^|;)\\s*' + name + '\\s*=\\s*([^;]+)');
+    if (res != undefined) {
+      return res.pop();
+    } else {
+      return null;
+    }
```

Comment:
you should use the `util` functions to do this instead of accessing directly

Reply:
used the StorageManager, thank you.